### PR TITLE
feature: support JSON parse statement #1746

### PR DESCRIPTION
**Pull request by amanPras**

### CODE REVIEW

File: server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/ElementaryItemNode.java [hunk 1]
```diff
@@ -43,7 +43,10 @@ public ElementaryItemNode(
       UsageFormat usageFormat,
       boolean redefines,
       boolean isBlankWhenZeroPresent,
-      boolean isSignClausePresent) {
```
Comment (Nurkambay) [inline server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/ElementaryItemNode.java:46]: I'd rather better to introduce 2nd constructor with additional parameters to keep the old code unchanged and allow to define of new parameters VIA a new constructor

File: server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/ElementaryNode.java [hunk 2]
```diff
@@ -36,25 +36,34 @@ public abstract class ElementaryNode extends VariableWithLevelNode
...
+          boolean redefines,
+          VariableType variableType,
+          boolean global,
+          boolean isBlankWhenZeroPresent,
+          boolean isSignClausePresent,
+          boolean isDynamicLength,
+          boolean isJustified,
+          boolean isUnBounded,
+          String picClause,
+          UsageFormat usageFormat) {
```
Comment (Nurkambay) [inline server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/ElementaryNode.java]: Same here

File: server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/ElementaryItemNode.java (hunk 1 above)
Comment (amanPras) [reply server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/ElementaryItemNode.java:46]: updated

File: server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/ElementaryNode.java (hunk 2 above)
Comment (amanPras) [reply server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/ElementaryNode.java]: updated

**Review by ishche — APPROVED** [review]

**Pull request merged**